### ISSUE TITLE
Port redirect only option is "Yes"

### ISSUE THREAD

**Issue by l0rd** [eclipse-che/che#17972]
### Describe the bug

<img width="493" alt="image" src="https://user-images.githubusercontent.com/606959/94277702-34fed480-ff4a-11ea-81ea-13a97c045f67.png">

We should have a "No" option as well and the message could be improved too: "...but this port is not in the workspace endpoints. Would you like to expose it as an external ingress and make it accessible from your browser?".

**Comment by tsmaeder** [comment]
And a "never ask me again" would be handy, as well. When developing Che-in-Che, it's really annoying to have to click it away each time I start the back end process.

**Comment by benoitf** [comment]
I think there are another issue to have the same design that is also used in VS Code Online codespaces now

TL;DR a widget instead of tons of annoying dialogs/popups

**Comment by l0rd** [comment]
We should never see this popup: when an endpoint is defined as `public: false` we should NOT ask if the user wants to redirect. That's one of the issues of the epic.

**Comment by benoitf** [comment]
@l0rd I think if there is no widget replacement we can't remove it.
Because, else you kill the other flow: if you start from scratch by cloning a nodejs express example that will listen on the port 3000 you want to have this example working and you don't want to add a new endpoint and restart your workspace at this point.

I agree on the popup removal but by having a widget replacement else the user will be stuck.

**Comment by l0rd** [comment]
> @l0rd I think if there is no widget replacement we can't remove it.
> Because, else you kill the other flow: if you start from scratch by cloning a nodejs express example that will listen on the port 3000 you want to have this example working and you don't want to add a new endpoint and restart your workspace at this point.
> 
> I agree on the popup removal but by having a widget replacement else the user will be stuck.

@benoitf I am not sure we are talking about the same thing. I have created a separate issue for that #17993 

> I think there are another issue to have the same design that is also used in VS Code Online codespaces now
> 
> TL;DR a widget instead of tons of annoying dialogs/popups

If there is another issue let's close this one.

**Comment by azatsarynnyy** [comment]
moving to `area/plugins` as the related functionality is in [`Ports Plugin`](https://github.com/eclipse/che-theia/tree/master/plugins/ports-plugin)
